_posts/2020-02-20-contractors.markdown

```diff
@@ -0,0 +1,213 @@
+| `last_name`                   | String            |           |          |         | The contractor's last name. This attribute is required for "Individual" contractors and will be ignored for "Business" contractors.
+| `email`                       | String            |           |    X     | null    | The contractor's email address. This attribute is optional for "Individual" contractors and will be ignored for "Business" contractors. This attribute should only be included in POST requests.
+| `self_onboarding`             | Boolean           |           |    X     | false   | Whether the contractor will complete onboarding in Gusto themselves, or if the administrator should complete their onboarding. It is recommended for contractors to self-onboard so that Gusto accounts are immediately created for them. If `self_onboarding` is true, then `email` is required. This attribute should only be included in POST requests.
+| `business_name`               | String            |           |          |         | The name of the contractor business. This attribute is required for "Business" contractors and will be ignored for "Individual" contractors.
+| `ein`                         | String            |           |    X     | null    | The employer identification number of the contractor business. This attribute is optional for "Business" contractors and will be ignored for "Individual" contractors.
+| `wage_type`                   | String            |           |          |         | The contractor's wage type, either "Fixed" or "Hourly".
+| `hourly_rate`                 | String            |           |    X     |   0.0   | The contractor's hourly rate. This attribute is required if the wage_type is "Hourly".
+| `address`                     | Object            |     X     |          |         | The contractor's home address.
+| `address:street_1`            | String            |     X     |          |         |
+| `address:street_2`            | String            |     X     |          |         |
+| `address:city`                | String            |     X     |          |         |
+| `address:state`               | String            |     X     |          |         |
+| `address:zip`                 | String            |     X     |          |         |
+
+## Get a contractor
+
+**HTTP Method**: `GET`
+
+**Endpoint**: `/v1/contractors/:contractor_id`
+
+**Returns**: The representation of a single contractor.
+
+#### Example Response
+
+```json
+{
+    "id": 7757515807594512,
+    "company_id": 7757616923763477,
+    "wage_type": "Fixed",
+    "is_active": false,
+    "version": "63859768485e218ccf8a449bb60f14ed",
+    "type": "Individual",
+    "first_name": "Kory",
+    "last_name": "Gottlieb",
+    "middle_initial": "P",
+    "business_name": null,
+    "ein": null,
+    "email": "keira.west@mckenzie.org",
+    "address": {
+        "id": 7757869449984763,
```

Comment:
Hot off the presses: Matan is changing up the address ID in [this PR](https://github.com/Gusto/zenpayroll/pull/56961). I would recommend removing the ID here. As he points out in that PR, address updates are in the context of an employee ID.

Reply:
In your other PR, I see you're removing the ID already, so this comment and future comments may not stand.

https://github.com/Gusto/zenpayroll/pull/56782/files#diff-dee11f4b093cd5165adbd852d4a6ad9dR25
https://github.com/Gusto/zenpayroll/pull/56782/files#diff-8eb8f152b8c41a6fad3826a41343d530L25
https://github.com/Gusto/zenpayroll/pull/56782/files#diff-9751faefc6d86987941bee5bceb0a732L162

Reply:
Thanks for surfacing it 🙏. 
The contractor use-case is a bit different (not a home address). but in general if we can avoid exposing ids in our apis, it'll make them significantly easier to change in the future.

Reply:
Ah, thanks for catching this! I removed address ID from the response (I had already chatted with Bo) and from the attributes table, but forgot to remove it from the example responses.